pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -107,12 +107,21 @@ func (l *Lifecycle) driverExists(obj *v3.KontainerDriver) bool {
 
 func (l *Lifecycle) download(obj *v3.KontainerDriver) (*v3.KontainerDriver, error) {
 	driver := drivers.NewKontainerDriver(obj.Spec.BuiltIn, obj.Name, obj.Spec.URL, obj.Spec.Checksum)
-	err := driver.Stage()
-	if err != nil {
-		return nil, err
-	}
 
-	v3.KontainerDriverConditionDownloaded.True(obj)
+	newObj, err := v3.KontainerDriverConditionDownloaded.Once(obj, func() (runtime.Object, error) {
```

Comment:
The condition `Downloaded` is very confusing. In the status map, it means start downloading. So you should update the status `Downloaded` before you actually begin downloading it via `driver.Stage()`
In the function `cond.Once(...)` the status Download is not set to you obj yet. So you need to change the logic as follow:
```
...
v3.KontainerDriverConditionDownloaded.True(obj)
obj, err := l.kontainerDrivers.Update(obj)
...
err = driver.Stage()
...
path, err := driver.Install()
...
v3.KontainerDriverConditionInstalled.True(obj)
...
obj, err := l.kontainerDrivers.Update(obj)
...
```